gnomad/assessment/summary_stats.py

```diff
@@ -151,3 +153,214 @@ def get_summary_counts(
+    mt = mt.filter_rows(filt_criteria)
+
+    if all_transcripts:
+        logger.info("Exploding transcript_consequences field...")
+        explode_field = "transcript_consequences"
+    else:
+        logger.info(
+            "Adding most severe (worst) consequence and expoding worst_csq_by_gene field..."
+        )
+        mt = process_consequences(mt)
+        explode_field = "worst_csq_by_gene"
+
+    if pre_loftee:
+        logger.info(f"Filtering to LoF consequences: {lof_csq_set}")
+        lof_cats = hl.literal(lof_csq_set)
+        criteria = lambda x: lof_cats.contains(
+            add_most_severe_consequence_to_consequence(x).most_severe_consequence
+        )
+    else:
+        logger.info("Filtering to LoF variants that pass LOFTEE with no LoF flags...")
+        criteria = lambda x: (x.lof == "HC") & hl.is_missing(x.lof_flags)
+
+    if additional_csq_set:
+        logger.info(f"Including these consequences: {additional_csq_set}")
+        additional_cats = hl.literal(additional_csq_set)
+        criteria &= lambda x: additional_cats.contains(
+            add_most_severe_consequence_to_consequence(x).most_severe_consequence
+        )
+
+    csqs = mt.vep[explode_field].filter(criteria)
+    mt = mt.select_rows(mt[freq_field], csqs=csqs)
+    mt = mt.explode_rows(mt.csqs)
+    annotation_expr = {
+        "gene_id": mt.csqs.gene_id,
+        "gene": mt.csqs.gene_symbol,
+        "indel": hl.is_indel(mt.alleles[0], mt.alleles[1]),
+        "most_severe_consequence": mt.csqs.most_severe_consequence,
+    }
+
+    if tx_ht:
+        logger.info("Adding transcript expression annotation...")
+        tx_annotation = get_tx_expression_expr(
+            mt.locus, mt.alleles, tx_ht, mt.csqs,
+        ).mean_proportion
+        annotation_expr["expressed"] = (
+            hl.case()
+            .when(tx_annotation >= high_expression_cutoff, "high")
+            .when(tx_annotation > low_expression_cutoff, "medium")
+            .when(hl.is_defined(tx_annotation), "low")
+            .default("missing")
+        )
+    else:
+        annotation_expr["transcript_id"] = mt.csqs.transcript_id
+        annotation_expr["canonical"] = hl.is_defined(mt.csqs.canonical)
+    mt = mt.annotate_rows(**annotation_expr)
+
+    return (
+        mt.group_rows_by(*list(annotation_expr.keys()))
+        .aggregate_rows(
+            n_sites=hl.agg.count(),
+            n_sites_array=hl.agg.array_sum(mt.freq.map(lambda x: hl.int(x.AC > 0))),
+            classic_caf=hl.agg.sum(mt[freq_field][freq_index].AF),
+            max_af=hl.agg.max(mt[freq_field][freq_index].AF),
+            classic_caf_array=hl.agg.array_sum(mt[freq_field].map(lambda x: x.AF)),
+        )
+        .aggregate_entries(
+            num_homs=hl.agg.count_where(mt.GT.is_hom_var()),
+            num_hets=hl.agg.count_where(mt.GT.is_het()),
+            defined_sites=hl.agg.count_where(hl.is_defined(mt.GT)),
+        )
+        .result()
+    )
```

Comment:
This looks great! There's a computation of `p` that I presume is coming in another PR?